charts/karpenter/templates/post-install-hook.yaml

```diff
@@ -12,7 +12,7 @@ metadata:
       {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
-  ttlSecondsAfterFinished: 0
+  ttlSecondsAfterFinished: {{ .Values.postInstallHook.ttlSecondsAfterFinished }}
```

Comment:
This mainly looks good, but I think we need to drop the annotation for `hook-succeeded` and `hook-failed` for the `helm.sh/hook-delete-policy`: https://helm.sh/docs/topics/charts_hooks/#hook-deletion-policies This will ensure that we are drive the clean-up process as part of the  `ttlSecondsAfterFinished`

Reply:
Makes sense to move these annotations out to a .Values.postInstallHook.annotations map then?

Reply:
Yeah, if user want that level of configuration that would make sense